Tella/Scenes/Resources/ViewModel/ResourcesViewModel.swift

```diff
@@ -0,0 +1,218 @@
+//
+//  ResourcesViewModel.swift
+//  Tella
+//
+//  Created by gus valbuena on 2/1/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+class ResourcesViewModel: ObservableObject {
+    private var appModel: MainAppModel
+    @Published var availableResources: [AvailableResourcesList] = []
+    @Published var downloadedResources: [DownloadedResourcesList] = []
+    @Published var isLoadingList: Bool = false
+    private var servers: [TellaServer] = []
+    private var cancellables: Set<AnyCancellable> = []
+    
+    var onShowResourceBottomSheet: ((String, String) -> Void)?
+
+    init(
+        mainAppModel: MainAppModel
+    ) {
+        self.appModel = mainAppModel
+        self.servers = mainAppModel.tellaData?.tellaServers.value ?? []
+
+        self.getAvailableForDownloadResources()
+        self.getDownloadedResources()
+    }
+
+    func getAvailableForDownloadResources() {
+        self.isLoadingList = true
+        self.availableResources = []
+
+        fetchAvailableResources()
+            .receive(on: DispatchQueue.main)
+            .sink(
+                receiveCompletion: { completion in
```

Comment:
I think it's better to separate the failure and success handling into two functions